question/classes/local/bank/view.php

```diff
@@ -0,0 +1,1191 @@
+     * breaking it up, validating the bits that are present, and returning them.
+     * If there is no subsort, then $subsort is returned as ''.
+     *
+     * @param string $sort the sort parameter to process.
+     * @return array array($colname, $subsort).
+     */
+    protected function parse_subsort($sort): array {
+        // Do the parsing.
+        if (strpos($sort, '-') !== false) {
+            list($colname, $subsort) = explode('-', $sort, 2);
+        } else {
+            $colname = $sort;
+            $subsort = '';
+        }
+        // Validate the column name.
+        $column = $this->get_column_type($colname);
+        if (!isset($column) || !$column->is_sortable()) {
+            for ($i = 1; $i <= self::MAX_SORTS; $i++) {
+                $this->baseurl->remove_params('qbs' . $i);
+            }
+            throw new \moodle_exception('unknownsortcolumn', '', $link = $this->baseurl->out(), $colname);
+        }
+        // Validate the subsort, if present.
+        if ($subsort) {
+            $subsorts = $column->is_sortable();
+            if (!is_array($subsorts) || !isset($subsorts[$subsort])) {
+                throw new \moodle_exception('unknownsortcolumn', '', $link = $this->baseurl->out(), $sort);
+            }
+        }
+        return array($colname, $subsort);
+    }
+
+    /**
+     * Initialise sort from parameters.
+     */
+    protected function init_sort_from_params(): void {
+        $this->sort = array();
+        for ($i = 1; $i <= self::MAX_SORTS; $i++) {
+            if (!$sort = optional_param('qbs' . $i, '', PARAM_TEXT)) {
+                break;
+            }
+            // Work out the appropriate order.
+            $order = 1;
+            if ($sort[0] == '-') {
+                $order = -1;
+                $sort = substr($sort, 1);
+                if (!$sort) {
+                    break;
+                }
+            }
+            // Deal with subsorts.
+            list($colname) = $this->parse_subsort($sort);
+            $this->requiredcolumns[$colname] = $this->get_column_type($colname);
+            $this->sort[$sort] = $order;
+        }
+    }
+
+    /**
+     * Sort to parameters.
+     * @param $sorts
+     * @return array
+     */
+    protected function sort_to_params($sorts): array {
+        $params = array();
+        $i = 0;
+        foreach ($sorts as $sort => $order) {
+            $i += 1;
+            if ($order < 0) {
+                $sort = '-' . $sort;
+            }
+            $params['qbs' . $i] = $sort;
+        }
+        return $params;
+    }
+
+    /**
+     * Default sort for question data.
+     * @return int[]
+     */
+    protected function default_sort(): array {
+        // Change required after implementing as plugins.
+        return array(
+                'core_question\local\bank\question_type_column' => 1,
+                //'core_question\local\bank\question_name_idnumber_tags_column-name' => 1
```

Comment:
This comment will fail in the codecheceker. should be something like

``// ...'core_question\local\bank\question_name_idnumber_tags_column-name' => 1``

Space after // and 2 dots if we want to start with lowercase.